### DISCUSSION TITLE
Feature request: Stats widget for embedding in web pages

### DISCUSSION THREAD

**Discussion by johnspurlock-skymethod** [skymethod/op3#21] [Ideas]
It would be great to have a little summary stats widget that podcasters using OP3 could easily embed on their web pages to show off their numbers / link to the show stats page.

_(This came up in a conversation with an existing podcaster using OP3)_

**Comment by Zesty0wl** [comment]
I just started thinking about this. I was thinking of writing a wordpress plugin to display op3 stats and also podcast episodes, along with stats. Basically a pretty podcast landing page to send people that also had stats.

My original plan went something like this...

1. Pull podcast feedxml for episode data, images, descriptions etc
2. Pull op3 data to get download stats and downloads per episode
3. Display grid of episodes with download stats for each

However, I got stuck. I was able to find this: https://op3.dev/api/1/shows/929c7e866c694ee08910a9cd8ce7df77/stats?token=preview07ce

Which gives me a dump of data for the show, which is great. However, I couldn't seem to figure out how to connect the episode data from here back to the show feed.xml, i.e. which exact episode is this data for?

  "episodeHourlyDownloads": {
    "90ae4594b2081fdaf51f0c44d91f0ef6d659e1e2b9474b41e3421230f64ce185": {
      "2025-01-13T08": 2,
      "2025-01-13T09": 2,
      "2025-01-13T11": 1,
      "2025-01-13T12": 1,
      "2025-01-13T13": 1,
      "2025-01-13T14": 1,
      "2025-01-13T15": 1,
      "2025-01-13T16": 1,
      "2025-01-13T18": 1,


Poking around on the current stats page I found that if I click the download button from the show page I get the following in tsv, which would be ideal, but it looks like it's calculated in memory on the client rather than stored anywhere which makes it difficult to get programaticallty?

episode_title	episode_pub_date	downloads_3_day	downloads_7_day	downloads_30_day	downloads_all_time	downloads_asof
The Making of Countdown to Extinction by Megadeth	2025-02-24T05:00:00.000Z	53	68		96	2025-03-10T00:00:00.000Z
The Making of Iowa by Slipknot	2025-02-17T05:00:00.000Z	57	77		97	2025-03-10T00:00:00.000Z
The Making of Stoosh by Skunk Anansie	2025-02-10T05:00:00.000Z	52	100		120	2025-03-10T00:00:00.000Z
The Making of Jagged Little Pill by Alanis Morissette	2025-02-03T05:00:00.000Z	41	61	89	93	2025-03-10T00:00:00.000Z
The Making of Rumours by Fleetwood Mac	2025-01-27T05:00:00.000Z	31	41	79	94	2025-03-10T00:00:00.000Z
The Making of the Fat of the Land by The Prodigy	2025-01-20T05:00:00.000Z	39	54	91	113	2025-03-10T00:00:00.000Z

I'm probably missing something simple, but would appreciate any pointers. I guess I am looking for a way to connect the op3 stats data to each individual podcast episode.
